ocaml/xapi/xapi_pgpu.ml

```diff
@@ -305,14 +305,14 @@ let set_GPU_group ~__context ~self ~value =
                   [gpu_type; "[" ^ String.concat ", " group_types ^ "]"]))
     )
 
-let get_remaining_capacity ~__context ~self ~vgpu_type =
-  match Xapi_pgpu_helpers.get_remaining_capacity_internal ~__context ~self ~vgpu_type with
+let get_remaining_capacity ~__context ~self ~vgpu_type ~pre_allocate_list =
```

Comment:
This is the entry point for the API, so adding the parameter here made it necessary to change the datamodel. Don't do that, and leave this function unchanged. Anything internal that calls `Xapi_pgpu.get_remaining_capacity` can just call `Xapi_pgpu_helpers.get_remaining_capacity`, which does the exact same thing and can receive the new argument. That's what "helpers" and "internal" functions are for.